### PR TITLE
SDIT-1264 Initial work for adjustments reconciliation

### PR DESCRIPTION
**Pull request by andymarke**
* Switch to OpenAPI generated classes for Adjustments
* Initial do nothing version of comparison reconcillation service
* Remove unused Mock resource

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/sentencing/SentencingAdjustmentsApiService.kt [hunk 1]
```diff
@@ -3,30 +3,28 @@ package uk.gov.justice.digital.hmpps.prisonertonomisupdate.sentencing
...
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.sentencing.adjustments.model.AdjustmentDto
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.sentencing.adjustments.model.LegacyAdjustment
 
 @Service
 class SentencingAdjustmentsApiService(private val sentenceAdjustmentsApiWebClient: WebClient) {
   companion object {
     const val LEGACY_CONTENT_TYPE = "application/vnd.nomis-offence+json"
   }
 
-  suspend fun getAdjustment(adjustmentId: String): AdjustmentDetails {
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/sentencing/SentencingAdjustmentsApiService.kt:14]: switch to generated class

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/sentencing/SentencingAdjustmentsService.kt [hunk 2]
```diff
@@ -63,12 +64,12 @@ class SentencingAdjustmentsService(
   private fun isDpsCreated(additionalInformation: AdditionalInformation) =
     additionalInformation.source != CreatingSystem.NOMIS.name
 
-  private suspend fun createTransformedAdjustment(adjustment: AdjustmentDetails) =
+  private suspend fun createTransformedAdjustment(adjustment: LegacyAdjustment) =
     CreateSentencingAdjustmentRequest(
-      adjustmentTypeCode = adjustment.adjustmentType,
+      adjustmentTypeCode = adjustment.adjustmentType.value,
       adjustmentDate = adjustment.adjustmentDate ?: LocalDate.now(),
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/sentencing/SentencingAdjustmentsService.kt:70]: enum in generated class

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/sentencing/SentencingReconciliationService.kt [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  }
+
+  suspend fun checkBookingIncentiveMatch(prisonerId: ActivePrisonerId): MismatchSentencingAdjustments? = runCatching {
+    val (nomisAdjustments, dpsAdjustments) = withContext(Dispatchers.Unconfined) {
+      async { nomisApiService.getAdjustments(prisonerId.bookingId) } to
+        async { adjustmentsApiService.getAdjustments(prisonerId.offenderNo) }
+    }.awaitBoth()
+
+    log.debug("will check adjustments for prisoner with ${prisonerId.offenderNo} booking: ${prisonerId.bookingId} : DPS has adjustments ${dpsAdjustments != null}")
+    return null
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/sentencing/SentencingReconciliationService.kt:33]: TODO - all the comparison stuff
doesn't do anything yet except calls both services - and this is never called by anything either

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/services/NomisApiService.kt [hunk 4]
```diff
@@ -284,6 +284,12 @@ class NomisApiService(@Qualifier("nomisApiWebClient") private val webClient: Web
       .retrieve()
       .awaitBody()
 
+  suspend fun getAdjustments(bookingId: Long): SentencingAdjustments =
+    webClient.get()
+      .uri("/prisoners/booking-id/{bookingId}/sentencing-adjustments", bookingId)
+      .retrieve()
+      .awaitBody()
+
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/services/NomisApiService.kt:292]: doesn't exist yet - bit will do next

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/services/NomisApiService.kt [hunk 5]
```diff
@@ -711,6 +717,10 @@ data class UpdateSentencingAdjustmentRequest(
 
 data class CreateSentencingAdjustmentResponse(val id: Long)
 
+data class SentencingAdjustments(
+  val dummy: String = "dummy",
+)
+
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/services/NomisApiService.kt:723]: place holder for real NOMIS API class once it exists

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/wiremock/SentencingAdjustmentsApiMockServer.kt [hunk 6]
```diff
@@ -44,6 +45,273 @@ class SentencingAdjustmentsApiMockServer : WireMockServer(WIREMOCK_PORT) {
...
 
+  fun stubAdjustmentsGet() {
+    stubFor(
+      get(urlPathMatching("/adjustments")).willReturn(
+        aResponse()
+          .withHeader("Content-Type", "application/json")
+          .withBody(
+            // language=json
+            """
+            [
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/wiremock/SentencingAdjustmentsApiMockServer.kt:56]: just cut and paste from Postman results for a Dev result, do something better once we need to start writing some tests

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/wiremock/SentencingAdjustmentsApiMockServer.kt [hunk 7]
```diff
@@ -71,6 +339,7 @@ class SentencingAdjustmentsApiMockServer : WireMockServer(WIREMOCK_PORT) {
               "adjustmentDays": $adjustmentDays,
               $startPeriodJson
               "bookingId": $bookingId,
+              "offenderNo": "A1234AA", 
               $sentenceSequenceJson
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/wiremock/SentencingAdjustmentsApiMockServer.kt:343]: now in the generated class as non-null so add to stubbed results

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/sentencing/SentencingAdjustmentsApiService.kt [hunk 8]
```diff
@@ -3,30 +3,28 @@ package uk.gov.justice.digital.hmpps.prisonertonomisupdate.sentencing
...
+  suspend fun getAdjustment(adjustmentId: String): LegacyAdjustment {
     return sentenceAdjustmentsApiWebClient.get()
       .uri("/legacy/adjustments/{adjustmentId}", adjustmentId)
       .header("Content-Type", LEGACY_CONTENT_TYPE)
       .retrieve()
       .awaitBody()
   }
-}
 
-data class AdjustmentDetails(
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/sentencing/SentencingAdjustmentsApiService.kt:23]: replaced with generated DTO

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**